src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1040,7 +1064,8 @@ struct at_dk_death_and_decay : AreaTriggerAI
 
     void OnUnitExit(Unit* unit) override
     {
-        unit->RemoveAurasDueToSpell(SPELL_DK_UNHOLY_GROUND_HASTE);
+        if (!unit->HasAura(SPELL_DK_CLEAVING_STRIKES))
+            unit->RemoveAurasDueToSpell(SPELL_DK_UNHOLY_GROUND_HASTE);
```

Comment:
shouldn't you modify remaining duration of SPELL_DK_DEATH_AND_DECAY_INCREASE_TARGETS on exit to 4s? (AuraEffect::GetAmount from EFFECT_3 of 316916)

Reply:
done